mkt/langpacks/serializers.py

```diff
@@ -36,7 +39,11 @@ def restore_object(self, attrs, instance=None):
         try:
             return LangPack.from_upload(attrs['upload'], instance=instance)
         except serializers.ValidationError, e:
-            raise exceptions.ParseError(e.messages)
+            raise ParseError(e.messages)
+        except SigningError, e:
+            # A SigningError could mean the package is corrupted but it often
+            # means something is wrong on our end, so return a 503.
+            raise ServiceUnavailable([e.message])
```

Comment:
At the end of the deserialization process. It's the method that creates (or updates) the model instance, which is why it's the one we are overriding here, since we want a completely custom way of creating our model instances.